### ISSUE TITLE
Some tests in `test_client_resource.py` run asynchronous operations without waiting for them to complete

### ISSUE THREAD

**Issue by exarkun** [PrivateStorageio/ZKAPAuthorizer#422]
These tests use `create_pumper` from Autobahn which returns an object with a `stop` method which returns a `Deferred`.  These tests uses `testtools.TestCase.addCleanup` to call the pumper `stop` method.  However, `testtools.TestCase.addCleanup` doesn't wait on `Deferred` return values unless `run_tests_with` is set to an `AsynchronousDeferredRunTest` instance - which these tests do not do.
